### PR TITLE
Mapbox GL Native v1.3.0

### PR DESCRIPTION
**Pull request by 1ec5**
Upgraded Mapbox GL Native from release-unicorn to [v1.3.0](https://github.com/mapbox/mapbox-gl-native/releases/tag/maps-v1.3.0):

 * Added the `in` expression function for testing whether a value is included in an array expression or whether a string is a substring of another string. Use this function in expressions in style JSON or with the `MGL_FUNCTION()` syntax in an `NSExpression` format string. (mapbox/mapbox-gl-native#16162)
* Added the `within` expression function for testing whether the evaluated feature lies within the given GeoJSON object. Use this function in expressions in style JSON or with the `MGL_FUNCTION()` syntax in an `NSExpression` format string. (mapbox/mapbox-gl-native#16157)
* Added the `MGLLineStyleLayer.lineSortKey` and `MGLFillStyleLayer.fillSortKey` properties. (#179, mapbox/mapbox-gl-native#16194, mapbox/mapbox-gl-native#16220)
* Icons and labels near tile boundaries draw more consistently. (mapbox/mapbox-gl-native#16159)
* Fixed an issue where an `MGLSymbolStyleLayer.lineDashPattern` value of `{1, 0}` resulted in hairline gaps. (mapbox/mapbox-gl-native#16202)
* Fixed issues where an offline pack would stop downloading before completion. (mapbox/mapbox-gl-native#16230, mapbox/mapbox-gl-native#16240)
* When an offline pack encounters an HTTP 404 error, the `MGLOfflinePackUserInfoKeyError` user info key of the `MGLOfflinePackErrorNotification` now indicates the resource that could not be downloaded. (mapbox/mapbox-gl-native#16240)
* Fixed a memory leak when zooming with any options enabled in the `MGLMapView.debugMask` property. (mapbox/mapbox-gl-native#15179)

Depends on mapbox/mapbox-gl-native#16242.

/cc @mapbox/maps-ios @mapbox/gl-native

### CODE REVIEW

**Comment by 1ec5** [comment]
iOS release builds are failing to link:

```
❌  ld: library not found for -lmbgl-vendor-icu
```

**Comment by 1ec5** [comment]
Locally, after cleaning up framework references in the iOS project targets, I’m seeing some concerning linker warnings too:

```
ld: warning: ignoring file /path/to/mapbox-gl-native-ios/build/ios/Debug-iphoneos/libmbgl-core.a, building for iOS-armv7 but attempting to link with file built for iOS-arm64
ld: warning: ignoring file /path/to/mapbox-gl-native-ios/build/ios/Debug-iphoneos/libmbgl-vendor-icu.a, building for iOS-armv7 but attempting to link with file built for iOS-arm64
```

**Comment by 1ec5** [comment]
mapbox/mapbox-gl-native#16158 increased the minimum deployment target from iOS 9.0 to iOS 12.0. This makes it incompatible with the iOS map SDK, which still deploys backward to iOS 9.0. It isn’t particularly obvious, but the linker errors and warnings above are due to this minimum deployment target mismatch, because [iOS 11 dropped the armv7 architecture](https://pspdfkit.com/guides/ios/current/configuration-issues/app-fails-to-archive/#architecture-mismatch).

**Comment by 1ec5** [comment]
mapbox/mapbox-gl-native#16031 attempted to port #140, but it introduced an unused file and omitted the MGLMapView integrations on both iOS and macOS, so I manually re-ported those changes.

**Comment by 1ec5** [comment]
> mapbox/mapbox-gl-native#16158 increased the minimum deployment target from iOS 9.0 to iOS 12.0. This makes it incompatible with the iOS map SDK, which still deploys backward to iOS 9.0. It isn’t particularly obvious, but the linker errors and warnings above are due to this minimum deployment target mismatch, because [iOS 11 dropped the armv7 architecture](https://pspdfkit.com/guides/ios/current/configuration-issues/app-fails-to-archive/#architecture-mismatch).

This PR is blocked by the fix in mapbox/mapbox-gl-native#16242.

File: platform/macos/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +21,8 @@
 * Fixed an issue that caused `MGLTileSourceOptionMaximumZoomLevel` to be ignored when setting `MGLTileSource.configurationURL`. ([#15581](https://github.com/mapbox/mapbox-gl-native/pull/15581))
 * Fixed an issue where `MGLSymbolStyleLayer.symbolSortKey` could sort text and icons incorrectly. ([#16023](https://github.com/mapbox/mapbox-gl-native/pull/16023))
 * Fixed an issue where style layers backed by a shape source could flicker when transitioning between styles. ([#15907](https://github.com/mapbox/mapbox-gl-native/pull/15907), [#15941](https://github.com/mapbox/mapbox-gl-native/pull/15941))
+* Icons and labels near tile boundaries draw more consistently. ([#16159](https://github.com/mapbox/mapbox-gl-native/pull/16159))
```
Comment (chloekraw) [inline platform/macos/CHANGELOG.md]: This change only affects `MapMode::Tile`

**Comment by 1ec5** [comment]
I’ve upgraded to mapbox-gl-native v1.3.0 and made the changes required by mapbox/mapbox-gl-native#16238.

**Comment by 1ec5** [comment]
These tests are failing:

* testOfflineQueryParameterIsAddedForOfflineResource, ((foundCount == 4) is true) failed
  /Users/distiller/project/platform/darwin/test/MGLResourceTests.mm:81
  ```

        XCTAssert(foundCount == 4);
  #else
  ```

* testOfflineQueryParameterIsAddedForOfflineResourceForChina, ((foundCount == 4) is true) failed
  /Users/distiller/project/platform/darwin/test/MGLResourceTests.mm:81
  ```

        XCTAssert(foundCount == 4);
  #else
  ```

**Pull request merged**